src/integration-test/java/org/spongepowered/mod/test/integration/regression/schematic/PlacingSchematicTest.java

```diff
@@ -0,0 +1,161 @@
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.mod.test.integration.regression.schematic;
+
+import static org.junit.Assert.assertEquals;
+
+import com.flowpowered.math.vector.Vector3i;
+import net.minecraftforge.fml.common.registry.GameRegistry;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.persistence.DataFormats;
+import org.spongepowered.api.data.persistence.DataTranslators;
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.entity.living.player.gamemode.GameModes;
+import org.spongepowered.api.event.block.ChangeBlockEvent;
+import org.spongepowered.api.world.BlockChangeFlags;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.World;
+import org.spongepowered.api.world.WorldArchetype;
+import org.spongepowered.api.world.WorldArchetypes;
+import org.spongepowered.api.world.extent.ArchetypeVolume;
+import org.spongepowered.api.world.schematic.PaletteTypes;
+import org.spongepowered.api.world.schematic.Schematic;
+import org.spongepowered.api.world.storage.WorldProperties;
+import org.spongepowered.common.regression.registry.Test1Tile;
+import org.spongepowered.common.regression.registry.Test2Tile;
+import org.spongepowered.common.regression.registry.Test3Tile;
+import org.spongepowered.common.regression.registry.TileEntityRegistrationTest;
+import org.spongepowered.mctester.api.junit.MinecraftRunner;
+import org.spongepowered.mctester.internal.BaseTest;
+import org.spongepowered.mctester.internal.event.StandaloneEventListener;
+import org.spongepowered.mctester.junit.TestUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.zip.GZIPInputStream;
+
+/**
+ *
+ */
+@RunWith(MinecraftRunner.class)
+public class PlacingSchematicTest extends BaseTest {
+
+
+    // The origin point where to apply all schematics.
+    public static final Vector3i ORIGIN = new Vector3i(0, 64, 0);
+    public static final Vector3i OFFSET = new Vector3i(4, 2, 4);
+
+
+    public PlacingSchematicTest(TestUtils testUtils) {
+        super(testUtils);
+    }
+
+    @BeforeClass
```

Comment:
@Aaron1011 btw,  this does work quite well, runs before the class is initialized, and the tile entities are being registered with forge.